pkg/types/chain_reader.go

```diff
@@ -105,10 +109,19 @@ type Sequence struct {
 }
 
 type BoundContract struct {
-	Address string
-	Name    string
+	Address  string
```

Comment:
Related to the comment above, maybe we don't need a method field, but can instead have a method on BoundContract for creating a key, so something like 
`GetLatestValue(ctx, bc.ReadID("key"), primitives.Finalized, params, returnVal)`

bc.ReadID("key")

Reply:
Alternatively if its easier or better to pass in `BoundContract` instead of a string identifiet, method field can be Unexported and hidden from the constructor and  `bc.ReadID("key")` could return a BoundContract with populated method field